### ISSUE TITLE
Список достижений

### ISSUE THREAD

**Issue by grihabor** [grihabor/garbage-clicker#56]
У достижения есть 3 параметра:

- Название
- Описание
- Условие, при котором достижение получается

Нужно продумать список достижений. Я предлагаю использовать интересные факты как описание достижений

**Comment by grihabor** [comment]
#54 Лучше связать достижения с названиями автоматов или названиями улучшений

**Comment by grihabor** [comment]
например у автомата сделать поле `achievements`

**Comment by grihabor** [comment]
типа 
```json
[{
  "name": "Мусорка",
  "achievements": [{
    "onCount": 5,
    "name": "Мусорный барон",
    "description": "Факт о мусорках"
  }, {

  }]
}, {

}]
```

**Comment by grihabor** [comment]
@RedSnail жду комментарии

**Comment by GeorgyBuzhenov** [comment]
Достижения на гугл диске

**Comment by RedSnail** [comment]
@grihabor мб сделать как-то так: в json как-то записать его условие выполнения в виде строки (чтобы таким образом дизайнеры могли без нашего участия их редачить), и, например, во втором потоке регулярно проверять достижения на выполнение

**Comment by RedSnail** [comment]
скажем, сделать отдельные выражения для того, чтобы отслеживать различные вещи
пример
"COUNT(factory) > 8"
"TOTAL_MONEY > 100500"
ну итд

**Comment by RedSnail** [comment]
это просто гибче -- мы сможем добавить туда какие угодно выражения -- от пресловутых count'ов до суммарной мощности; количества денег, потраченных на автоматы; количества часов в игре ну итд

**Comment by grihabor** [comment]
Не очень хочется такое парсить

**Comment by grihabor** [comment]
Давай пока разберемся, как это работает в Google Play Game Services

**Comment by RedSnail** [comment]
да, я +- разобрался и активация улучшений всё равно должна происходить из кода

**Comment by RedSnail** [comment]
в принципе всё то же самое, что мы бы делали без google play, только кода меньше
